### ISSUE TITLE
proposal: add Len/Length method on Result.

### ISSUE THREAD

**Issue by wI2L** [tidwall/gjson#127]
Hello @tidwall 

I'm back with another change proposal for you.
Last time we introduced the [change](https://github.com/tidwall/gjson/pull/11) on the `Array` method.

This time, I'd like to propose the introduction of a new method for the `Result` type, `Length` (or `Len`).
This method is obviously targeted at JSON arrays, but could also work for strings and objects, ie: return length of string (bytes count), and the number of key/value pairs of an object. The behavior for numbers and boolean values still has to be defined. It could panic (like the reflect package does), or simply return 0, which seems preferable to me, and in agreement with the current codebase.

The rationale behind this is to provide a clear way within the API to count the number of elements in an array without having to call `Array` explicitly. Typically, one would want to pre-allocate a destination Go slice/array that has the same capacity than the source, call `ForEach` to iterate over the members of a JSON array/object and abort prematurely for whatever reason (no time is spent parsing/allocating values that won't be necessary).

I think the implementation could be a lightweight/simplified version of the `ForEach` method, without key/value parsing.

W.

PS: I am willing to tackle this myself if you want me to.

**Comment by tidwall** [comment]
Hi @wI2L,

The `#` path component is intended to return the length of an array. It works on a json string or a `gjson.Result`.

```go
gjson.Get(`[1,2,3,4]`, "#")
gjson.Get(`[1,2,3,4]`).Get("#")
```

Would this work in your case?

**Comment by wI2L** [comment]
Very interresting. I indeed thought of using the `#` path parameter, but I didn't knew that it could be used  directly on a `Result`, and this changes everything, because this is precisely the usecase I have (ie. not having to deal with the full path).

So, coming back to the original proposal, there's maybe something to improve there. Either the README could use a mention of that, or the `Length` method I proposed could be an alias for this on the result ? I like to think that it is easier to use explicit API methods when dealing with a `gjson.Result` instance instead of the "path spec", but that is just my own feeling.
